### ISSUE TITLE
Polish Translation Progress

### ISSUE THREAD

**Issue by iliakan** [javascript-tutorial/pl.javascript.info#1]
## Maintainer List

@aleksanderkrauze

## For New Translators

<details><summary><b>Please read this first (click to open)</b></summary>
<p>
To translate an article:

1. Check that no one else has claimed your article in the checklist below.
2. Comment below with the title of the article that you would like to translate, e.g. `An Introduction to JavaScript`. 
    - **Please take only one article at a time**.
3. Fork this repo, translate the article in your fork and submit a pull request!
    - The pull request title should be same as the article, e.g. `An Introduction to JavaScript` (just like comment)

Please be prompt with your translations! If you find find that you can't commit any more, let the maintainers know so they can assign the page to someone else.
</p>
</details>

## For Maintainers

<details><summary><b>Click to open</b></summary>
<p>
Please let others know what you do, on community boards and chats, invite them to join. Translations become better if more people see them.

For teams we suggest that an article has 2 reviews to be merged. 

Translalions are tracked below, like this:

* [ ] [Home Page](url) (@iliakan) #1

There's a helper bot to track translations, you can read more about it at <https://github.com/javascript-tutorial/translate/edit/master/BOT.md>.

If something doesn't work right, please contact @iliakan. 

</p>
</details>

Only maintainers can check/uncheck items below. If you're not, please write in a comment what you take to translate.

### An introduction

* [ ] [An Introduction to JavaScript](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/01-getting-started/1-intro)
* [ ] [Manuals and specifications](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/01-getting-started/2-manuals-specifications)
* [ ] [Code editors](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/01-getting-started/3-code-editors)
* [ ] [Developer console](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/01-getting-started/4-devtools)

### JavaScript Fundamentals

* [ ] [Hello, world!](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/01-hello-world)
* [ ] [Code structure](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/02-structure)
* [ ] [The modern mode, "use strict"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/03-strict-mode)
* [ ] [Variables](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/04-variables)
* [ ] [Data types](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/05-types)
* [ ] [Type Conversions](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/06-type-conversions)
* [ ] [Operators](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/07-operators)
* [ ] [Comparisons](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/08-comparison)
* [ ] [Interaction: alert, prompt, confirm](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/09-alert-prompt-confirm)
* [ ] [Conditional operators: if, '?'](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/10-ifelse)
* [ ] [Logical operators](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/11-logical-operators)
* [ ] [Loops: while and for](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/12-while-for)
* [ ] [The "switch" statement](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/13-switch)
* [ ] [Functions](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/14-function-basics)
* [ ] [Function expressions and arrows](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/15-function-expressions-arrows)
* [ ] [JavaScript specials](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/16-javascript-specials)

### Code quality

* [ ] [Debugging in Chrome](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/03-code-quality/01-debugging-chrome)
* [ ] [Coding Style](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/03-code-quality/02-coding-style)
* [ ] [Comments](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/03-code-quality/03-comments)
* [ ] [Ninja code](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/03-code-quality/04-ninja-code)
* [ ] [Automated testing with mocha](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/03-code-quality/05-testing-mocha)
* [ ] [Polyfills](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/03-code-quality/06-polyfills)

### Objects: the basics

* [ ] [Objects](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/04-object-basics/01-object)
* [ ] [Garbage collection](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/04-object-basics/02-garbage-collection)
* [ ] [Symbol type](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/04-object-basics/03-symbol)
* [ ] [Object methods, "this"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/04-object-basics/04-object-methods)
* [ ] [Object to primitive conversion](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/04-object-basics/05-object-toprimitive)
* [ ] [Constructor, operator "new"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/04-object-basics/06-constructor-new)

### Data types

* [ ] [Methods of primitives](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/01-primitives-methods)
* [ ] [Numbers](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/02-number)
* [ ] [Strings](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/03-string)
* [ ] [Arrays](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/04-array)
* [ ] [Array methods](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/05-array-methods)
* [ ] [Iterables](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/06-iterable)
* [ ] [Map, Set, WeakMap and WeakSet](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/07-map-set-weakmap-weakset)
* [ ] [Object.keys, values, entries](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/08-keys-values-entries)
* [ ] [Destructuring assignment](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/09-destructuring-assignment)
* [ ] [Date and time](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/10-date)
* [ ] [JSON methods, toJSON](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/05-data-types/11-json)

### Advanced working with functions

* [ ] [Recursion and stack](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/01-recursion)
* [ ] [Rest parameters and spread operator](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/02-rest-parameters-spread-operator)
* [ ] [Closure](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/03-closure)
* [ ] [The old "var"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/04-var)
* [ ] [Global object](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/05-global-object)
* [ ] [Function object, NFE](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/06-function-object)
* [ ] [The "new Function" syntax](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/07-new-function)
* [ ] [Scheduling: setTimeout and setInterval](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/08-settimeout-setinterval)
* [ ] [Decorators and forwarding, call/apply](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/09-call-apply-decorators)
* [ ] [Function binding](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/10-bind)
* [ ] [Currying and partials](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/11-currying-partials)
* [ ] [Arrow functions revisited](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/12-arrow-functions)

### Object properties configuration

* [ ] [Property flags and descriptors](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/07-object-properties/01-property-descriptors)
* [ ] [Property getters and setters](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/07-object-properties/02-property-accessors)

### Prototypes, inheritance

* [ ] [Prototypal inheritance](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/08-prototypes/01-prototype-inheritance)
* [ ] [F.prototype](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/08-prototypes/02-function-prototype)
* [ ] [Native prototypes](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/08-prototypes/03-native-prototypes)
* [ ] [Prototype methods, objects without __proto__](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/08-prototypes/04-prototype-methods)

### Classes

* [ ] [Class basic syntax](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/01-class)
* [ ] [Class inheritance](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/02-class-inheritance)
* [ ] [Static properties and methods](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/03-static-properties-methods)
* [ ] [Private and protected properties and methods](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/04-private-protected-properties-methods)
* [ ] [Extending built-in classes](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/05-extend-natives)
* [ ] [Class checking: "instanceof"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/06-instanceof)
* [ ] [Mixins](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/09-classes/07-mixins)

### Error handling

* [ ] [Error handling, "try..catch"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/10-error-handling/1-try-catch)
* [ ] [Custom errors, extending Error](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/10-error-handling/2-custom-errors)

### Promises, async/await

* [ ] [Introduction: callbacks](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/01-callbacks)
* [ ] [Promise](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/02-promise-basics)
* [ ] [Promises chaining](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/03-promise-chaining)
* [ ] [Error handling with promises](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/04-promise-error-handling)
* [ ] [Promise API](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/05-promise-api)
* [ ] [Promisification](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/06-promisify)
* [ ] [Microtasks](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/07-microtask-queue)
* [ ] [Async/await](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/11-async/08-async-await)

### Generators, advanced iteration

* [ ] [Generators](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/12-generators-iterators/1-generators)
* [ ] [Async iterators and generators](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/12-generators-iterators/2-async-iterators-generators)

### Modules

* [ ] [Modules, introduction](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/13-modules/01-modules-intro)
* [ ] [Export and Import](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/13-modules/02-import-export)
* [ ] [Dynamic imports](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/13-modules/03-modules-dynamic-imports)

### Miscellaneous

* [ ] [Proxy and Reflect](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/99-js-misc/01-proxy)
* [ ] [Eval: run a code string](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/99-js-misc/02-eval)

### Document

* [ ] [Browser environment, specs](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/01-browser-environment)
* [ ] [DOM tree](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/02-dom-nodes)
* [ ] [Walking the DOM](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/03-dom-navigation)
* [ ] [Searching: getElement*, querySelector*](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/04-searching-elements-dom)
* [ ] [Node properties: type, tag and contents](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/05-basic-dom-node-properties)
* [ ] [Attributes and properties](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/06-dom-attributes-and-properties)
* [ ] [Modifying the document](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/07-modifying-document)
* [ ] [Styles and classes](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/08-styles-and-classes)
* [ ] [Element size and scrolling](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/09-size-and-scroll)
* [ ] [Window sizes and scrolling](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/10-size-and-scroll-window)
* [ ] [Coordinates](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/1-document/11-coordinates)

### Introduction to Events

* [ ] [Introduction to browser events](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/2-events/01-introduction-browser-events)
* [ ] [Bubbling and capturing](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/2-events/02-bubbling-and-capturing)
* [ ] [Event delegation](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/2-events/03-event-delegation)
* [ ] [Browser default actions](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/2-events/04-default-browser-action)
* [ ] [Dispatching custom events](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/2-events/05-dispatch-events)

### UI Events

* [ ] [Mouse events basics](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/3-event-details/1-mouse-events-basics)
* [ ] [Moving: mouseover/out, mouseenter/leave](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/3-event-details/3-mousemove-mouseover-mouseout-mouseenter-mouseleave)
* [ ] [Drag'n'Drop with mouse events](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/3-event-details/4-mouse-drag-and-drop)
* [ ] [Keyboard: keydown and keyup](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/3-event-details/5-keyboard-events)
* [ ] [Scrolling](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/3-event-details/8-onscroll)

### Forms, controls

* [ ] [Form properties and methods](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/4-forms-controls/1-form-elements)
* [ ] [Focusing: focus/blur](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/4-forms-controls/2-focus-blur)
* [ ] [Events: change, input, cut, copy, paste](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/4-forms-controls/3-events-change-input)
* [ ] [Form submission: event and method submit](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/4-forms-controls/4-forms-submit)

### Document and resource loading

* [ ] [Page: DOMContentLoaded, load, beforeunload, unload](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/5-loading/01-onload-ondomcontentloaded)
* [ ] [Scripts: async, defer](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/5-loading/02-script-async-defer)
* [ ] [Resource loading: onload and onerror](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/5-loading/03-onload-onerror)

### Miscellaneous

* [ ] [Mutation observer](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/99-ui-misc/01-mutation-observer)
* [ ] [Selection and Range](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/99-ui-misc/02-selection-range)
* [ ] [Event loop: microtasks and macrotasks](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/2-ui/99-ui-misc/03-event-loop)

## Frames and windows

* [ ] [Popups and window methods](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/3-frames-and-windows/01-popup-windows)
* [ ] [Cross-window communication](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/3-frames-and-windows/03-cross-window-communication)
* [ ] [The clickjacking attack](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/3-frames-and-windows/06-clickjacking)

## Binary data, files

* [ ] [ArrayBuffer, binary arrays](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/4-binary/01-arraybuffer-binary-arrays)
* [ ] [TextDecoder and TextEncoder](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/4-binary/02-text-decoder)
* [ ] [Blob](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/4-binary/03-blob)
* [ ] [File and FileReader](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/4-binary/04-file)

## Network requests

* [ ] [Fetch: Basics](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/01-fetch-basics)
* [ ] [Fetch: Download progress](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/02-fetch-progress)
* [ ] [Fetch: Abort](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/03-fetch-abort)
* [ ] [Fetch: Cross-Origin Requests](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/04-fetch-crossorigin)
* [ ] [Fetch API](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/05-fetch-api)
* [ ] [URL objects](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/06-url)
* [ ] [XMLHttpRequest](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/07-xmlhttprequest)
* [ ] [WebSocket](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/08-websocket)
* [ ] [Server Sent Events](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/5-network/09-server-sent-events)

## Storing data in the browser

* [ ] [Cookies, document.cookie](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/6-data-storage/01-cookie)
* [ ] [LocalStorage, sessionStorage](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/6-data-storage/02-localstorage)
* [ ] [IndexedDB](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/6-data-storage/03-indexeddb)

## Animation

* [ ] [Bezier curve](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/7-animation/1-bezier-curve)
* [ ] [CSS-animations](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/7-animation/2-css-animations)
* [ ] [JavaScript animations](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/7-animation/3-js-animation)

## Web components

* [ ] [From the orbital height](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/1-webcomponents-intro)
* [ ] [Custom elements](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/2-custom-elements)
* [ ] [Shadow DOM](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/3-shadow-dom)
* [ ] [Template element](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/4-template-element)
* [ ] [Shadow DOM slots, composition](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/5-slots-composition)
* [ ] [Shadow DOM styling](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/6-shadow-dom-style)
* [ ] [Shadow DOM and events](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/8-web-components/7-shadow-dom-events)

## Regular expressions

* [ ] [Patterns and flags](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/01-regexp-introduction)
* [ ] [Methods of RegExp and String](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/02-regexp-methods)
* [ ] [Character classes](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/03-regexp-character-classes)
* [ ] [Escaping, special characters](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/04-regexp-escaping)
* [ ] [Sets and ranges [...]](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/05-regexp-character-sets-and-ranges)
* [ ] [Quantifiers +, *, ? and {n}](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/07-regexp-quantifiers)
* [ ] [Greedy and lazy quantifiers](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/08-regexp-greedy-and-lazy)
* [ ] [Capturing groups](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/09-regexp-groups)
* [ ] [Backreferences in pattern: \n and \k](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/10-regexp-backreferences)
* [ ] [Alternation (OR) |](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/11-regexp-alternation)
* [ ] [String start ^ and finish $](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/12-regexp-anchors)
* [ ] [Multiline mode, flag "m"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/13-regexp-multiline-mode)
* [ ] [Lookahead and lookbehind](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/14-regexp-lookahead-lookbehind)
* [ ] [Infinite backtracking problem](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/15-regexp-infinite-backtracking-problem)
* [ ] [Unicode: flag "u"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/20-regexp-unicode)
* [ ] [Unicode character properties \p](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/21-regexp-unicode-properties)
* [ ] [Sticky flag "y", searching at position](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/9-regular-expressions/22-regexp-sticky)

**Comment by justdzasta** [comment]
I would like to translate the `The modern mode, "use strict"`

**Comment by aleksanderkrauze** [comment]
OK, go ahead @justdzasta

**Comment by Verthon** [comment]
I would like to translate the `Data types`

**Comment by JanKesek** [comment]
i can translate Document in Browser: Document, Events, Interfaces, starting with Browser environment, specs

**Comment by random-passenger** [comment]
I would like to translate `Hello World!`.

**Comment by effhAVE** [comment]
`Variables`

**Comment by bartek7** [comment]
`Code editors`

**Comment by bartek7** [comment]
`Developer console`

**Comment by bartek7** [comment]
`Hello, world!`

**Comment by bartek7** [comment]
Ok, so I saw there is someone who wants to translate these topics but is there any progress?
`The modern mode, "use strict"`, `Variables`

I'll translate now `Code structure` but then I'll translate above topics.

**Comment by bartek7** [comment]
I've created new pull requests for my translated pages. I've made some mistakes in previous (so I closed them). For now I will translate `Code structure`. Come on guys! We need this page in polish community!

**Comment by bartek7** [comment]
@justdzasta wrote that want to translate `The modern mode, "use strict"`. But after 9 days here is nothing in pull request on that. So I'll translate it and create pull request as soon as possible.

**Comment by Verthon** [comment]
> Interaction: alert, prompt, confirm

**Comment by JanKesek** [comment]
`Generators`

**Comment by bartek7** [comment]
The next for me is `Operators` and `Comparisions`

**Comment by bartek7** [comment]
@iliakan @aleksanderkrauze Do you need more "Maintainers"? I can be one. And there are 9 pull requests. We should review pull requests more often.

**Comment by farynski** [comment]
If no one has started on the **Objects** topic yet I can translate the first three subtopics : _Objects_, _Garbage collection_ and S_ymbol type_.

**Comment by bartek7** [comment]
@iliakan @aleksanderkrauze Can we create a Discord or something like that to communicate with each other?

**Comment by matwjablonski** [comment]
Hi, can I join to this project and help you? :)

**Comment by matwjablonski** [comment]
I will take `Prototypes, inheritance` :)

**Comment by Versoo** [comment]
Conditional operators: if, '?'

**Comment by mitroc** [comment]
Hi guys, is this project still alive? 11 PRs, some from July...
Do we have some Slack channel or something?

**Comment by iliakan** [comment]
@bartek7 @mitroc should I assign you as maintainers? Looks like the original maintainer @aleksanderkrauze isn't online any more.

**Comment by mitroc** [comment]
@iliakan Me, a maintainer? Sounds like a VIP :). 
But seriously, thanks, but I'm not that good at JavaScript. I will gladly translate some simpler concepts, but I will not undertake to review those that I do not understand myself.

**Comment by iliakan** [comment]
The translation does not have maintainers now, as I see it.

**Comment by iliakan** [comment]
@jakubdrozdek Hey, great to hear from you! 

We really need a maintainer here. Added you, invite sent.

P.S. You might want to add the glossary link to the Readme.

**Comment by iliakan** [comment]
@jakubdrozdek We have a bot to ensure double-review of translations. When changes are requested, it adds a label, making PRs easier to filter.

The original bot idea was from react translate bot.

Until we have more reviewers, you can just merge instead of "approving changes". Still, if we get double reviews, that would make the translation quality better.

**Comment by jakubdrozdek** [comment]
@bartek7 @JanKesek @Verthon @effhAVE Your PRs are still on the list, waiting for a review or merge. Are you willing to correct any mistakes and will you help with the further translation? Or can I copy/paste your translation and mark the article as translated by you (on the list in this issue). I'm asking because your PRs were created in your forked repos and I can't modify them in any way.

**Comment by jakubdrozdek** [comment]
@iliakan Thanks for the explanation. I've got a question: how am I supposed to mark any items on the list as done? I don't have the permission to edit it, and all of the checkboxes are disabled. Any ideas?
In the React repo I had full access to the repo, so I could edit anything.

**Comment by iliakan** [comment]
@jakubdrozdek if a PR was created as a fork of this repo, you can pull the relevant branch and merge it locally, so there's no need to copy-paste.

Did you receive the github invite to join the org? If you did and joined, you have the access.

**Comment by jakubdrozdek** [comment]
You're right, it worked :-)

**Comment by mitroc** [comment]
I'll take `fetch`.

**Comment by jakubdrozdek** [comment]
@iliakan Any plans on hosting the `pl` domain soon?

**Comment by iliakan** [comment]
@jakubdrozdek there's no point in opening `pl` in its current almost zero-translated state.

**Comment by mitroc** [comment]
@jakubdrozdek Can we use polish here or in reviews comments?

**Comment by jakubdrozdek** [comment]
@mitroc Yeah, I think so. It will make our comments better, more precise and more readable, because we won't mix English and Polish. Let's stick to Polish in PRs, and use English in this thread, since it's more 'public' and serves as the progress indicator for people outside.

**Comment by mitroc** [comment]
Deal.

**Comment by jakubdrozdek** [comment]
Let's wait for PR reviews for max. 7 days to ensure smooth progress. This means that if somebody creates a PR, some other person reviews it and we still need +1 to pass the PR acceptance rules, I (or any other maintainer) will decide if we should merge it after the week or not. We don't want PRs to wait for months just because there are no active reviewers.

**Comment by mitroc** [comment]
Sure. And what about the PRs, that have been hanging here for months?

**Comment by jakubdrozdek** [comment]
I'll review them, correct and ask for review again :-)

**Comment by mitroc** [comment]
Oh my...

**Comment by iliakan** [comment]
@jakubdrozdek I guess, right now you can merge PRs single-handedly.

**Comment by jakubdrozdek** [comment]
@iliakan I guess you are right. I wanted to 'lure' some more maintainers by allowing them to review my translations, but I will probably gain more by simply pushing the translation progress forward.
What are your requirements for creating the `pl` domain? Do you want us to reach a particular milestone? Maybe I could prioritize some of the articles in order to reach out to the world sooner.

**Comment by iliakan** [comment]
@jakubdrozdek Surely more maintainers is better, especially with experience from React translation =)

Usually we publish as far as 50% is translated.

**Comment by iliakan** [comment]
At least part 1.

**Comment by mitroc** [comment]
`FormData`

**Comment by mitroc** [comment]
Guys, this might be a stupid question, but I've almost never made a pull request. Once one PR is in progress I can not make another PR to the same branch, so they are kind of separate? I see that the second PR, which I planned to name `FormData` was dragged into the first - `Fetch`m without even giving me the chance to name it...  :-/

**Comment by jakubdrozdek** [comment]
Yeah, this is because PR is an intent to merge some particular branch into another one. And if you create a PR with branch "X", and then add something else to the same branch, it will appear in the PR, too.

You should stick to those rules:
- before starting your work, checkout to master branch and pull latest changes - `git checkout master && git pull`
- create a new branch from master - `git checkout -b new-branch`
- now you may start translating the text, then commit your changes to the branch - `git commit -m "my commit message"`, and then `git push`
All of those actions can be made with a decent IDE, such as Visual Studio Code.
There might be some differences in your flow since you work on a fork of this repo. The main rule is that each PR should be made with a separate branch.

**Comment by mitroc** [comment]
Holy crap! Sounds like I screwed thins up. :), but really I made my last PR in 2017 and it was the third one ;). 
Anyway, @jakubdrozdek thanks for the guide, I'think it might be a good idea to pin your post to the repo description for posterity.

**Comment by sbielenica** [comment]
Hey, I'd like to translate _Comments_ from _Code quality_ section.

**Comment by jakubdrozdek** [comment]
Hello, @sbielenica ;-) Welcome aboard.

**Comment by mitroc** [comment]
`Fetch: Abort`

**Comment by stnStanley** [comment]
Logical operators

**Comment by TomaszWaszczyk** [comment]
Anyone translates https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/02-first-steps/16-arrow-functions-basics ?

**Comment by Katolus** [comment]
Hey guys, I would like to help. I can take a section like `Introduction to Events` if we can divide by sections. If not I will take an item each time a branch gets in. Up to you!

**Comment by piotrbednarski-hub** [comment]
I want to translate `Objects` is possible or not?

**Comment by Fjollsfinn** [comment]
Methods of primitives

**Comment by Katolus** [comment]
Introduction to browser events

**Comment by Fjollsfinn** [comment]
@kkrawczykpl @Katolus Would you like to work on translations with me ? I think that we still need more people to make some CRs. If you've finished any of translations please make a PR so I can check it.

**Comment by sculpt0r** [comment]
Nullish coalescing operator '??'

**Comment by sculpt0r** [comment]
Hi :)
I totally can't find "Nullish coalescing operator '??'"
It should be at `1-js/02-first-steps/12-nullish-coalescing-operator` but at 12 - there is `while-for`.

Am I doing sth wrong or this repo isn't synced with the original?

**Comment by Fjollsfinn** [comment]
@sculpt0r Thanks for your vigilance. This repo wasn't sync with the EN version. The 12-nullish-coalescing-operator was missing. I've updated it already. Pull from master branch and start working on 12-nullish-coalescing-operator on new feature branch please.
Good luck! :)

**Comment by Moviss** [comment]
`Loops: while and for`

**Comment by Fjollsfinn** [comment]
@Moviss In future if you want to be marked in translation progress automatically by bot, please comment like this: Loops: while and for without backticks :)

**Comment by Fjollsfinn** [comment]
The "switch" statement

**Comment by Moviss** [comment]
> @Moviss In future if you want to be marked in translation progress automatically by bot, please comment like this: Loops: while and for without backticks :)

Thank you - I will remeber :)

**Comment by Moviss** [comment]
Loops: while and for

**Comment by platypuswithoutk** [comment]
Arrow functions, the basics

**Comment by dyrpit** [comment]
Functions

**Comment by Null-altV** [comment]
Debugging in Chrome

**Comment by VibingCreator** [comment]
Error handling, "try..catch"

**Comment by dwolanin** [comment]
Objects

**Comment by kplich** [comment]
Function expressions

**Comment by TheTasak** [comment]
Numbers

**Comment by poplaw** [comment]
Object references and copying

**Comment by ArturKot95** [comment]
Coding Style

**Comment by Onxi95** [comment]
JavaScript specials

**Comment by covalt1985** [comment]
Object methods, "this"

**Comment by ostafinskim** [comment]
Arrays

**Comment by deivuss331** [comment]
Optional chaining '?.'

**Comment by iliakan** [comment]
@jakubdrozdek @Fjollsfinn are you here? Do you still contribute/review PRs?

**Comment by jakubdrozdek** [comment]
> @jakubdrozdek @Fjollsfinn are you here? Do you still contribute/review PRs?

No, sorry. I haven't looked here for a long while and I believe I won't continue my work here. Hope you find somebody else 🤞

**Comment by iliakan** [comment]
@deivuss331 @poplaw Welcome!

**Comment by ArturKot95** [comment]
I cant continue my work for now

**Comment by akowalska622** [comment]
I can try with pleasure!

**Comment by akowalska622** [comment]
Hey @iliakan, I wanted to review some PRs, but it seems like I can't do it for some reason :(
I mean there's no option in the interface to review, should I have some permissions?

**Comment by iliakan** [comment]
@akowalska622 did you accept the invite? It must have been sent to your email 11 days ago (right when I added you).

**Comment by akowalska622** [comment]
@iliakan I must have missed it! 
I found the mail with invitation, but unfortunately it expired after 7 days. Could you invite me again?
I'll check some PRs right away :D

**Comment by iliakan** [comment]
Try visiting https://github.com/javascript-tutorial/pl.javascript.info, do you see a notification with the invite above?
Or https://github.com/javascript-tutorial/pl.javascript.info/invitations.

**Comment by RafalMichniuk** [comment]
Strings

**Comment by RafalMichniuk** [comment]
Garbage collection

**Comment by suff42** [comment]
Iterables

**Comment by bartek7** [comment]
Error handling, "try...catch"

**Comment by bartek7** [comment]
Custom errors, extending Error

**Comment by riftmat** [comment]
Browser environment, specs

**Comment by khaaliee** [comment]
I would like to translate [The old "var"](https://github.com/javascript-tutorial/pl.javascript.info/blob/master/1-js/06-advanced-functions/04-var)

**Comment by ostafinskim** [comment]
Hi @iliakan
Due lack of time I need to drop my translate assignment of Array section. Cheers!

**Comment by bartosztrusinski** [comment]
Polyfills and transpilers
